packages/devtools_app/lib/src/service/vm_service_wrapper.dart

```diff
@@ -953,15 +957,48 @@ class VmServiceWrapper implements VmService {
     vmServiceCallCount = 0;
   }
 
+  /// Wraps a future with logs that at its start and finish.
+  ///
+  /// All logs from this run will have matching unique ids, so that they can
+  /// be associated together in the logs.
+  Future<T> _logWrappedFuture<T>(
+    String name,
+    Future<T> future,
+  ) async {
+    /// A unique id to add to each of the future's messages.
+    final futureLogId = Random().nextInt(1 << 30);
+
+    try {
+      _log.info('[$futureLogId]-trackFuture($name,...): Started');
+      final result = await future;
+      _log.info('[$futureLogId]-trackFuture($name,...): Succeeded');
+      return result;
+    } catch (error) {
+      _log.info(
+        '[$futureLogId]-trackFuture($name,...): Failed',
+        error,
+      );
+      rethrow;
+    }
+  }
+
   @visibleForTesting
   Future<T> trackFuture<T>(String name, Future<T> future) {
+    Future<T> localFuture = future;
+
+    if (_log.isLoggable(Level.INFO)) {
```

Comment:
Can we change this to check if the preference verboseLogging is enabled?

Reply:
I would prefer to leave this connected to Level.INFO since this is protecting _log.info(...) requests.

I will move this condition into the helper function so it is more clear that they are closely related.

Reply:
For now I've moved this to FINE as it is not something that we want to be logging to everyone :D